src/robot.js

```diff
@@ -355,21 +353,39 @@ class Robot {
   // filename - A String filename in path on the filesystem.
   //
   // Returns nothing.
-  loadFile (filepath, filename) {
+  async loadFile (filepath, filename) {
     const ext = path.extname(filename)
-    const full = path.join(filepath, path.basename(filename, ext))
+    const full = path.join(filepath, path.basename(filename))
 
     // see https://github.com/hubotio/hubot/issues/1355
     if (['.js', '.mjs', '.coffee'].indexOf(ext) == -1) { // eslint-disable-line
```

Comment:
Should we consider .cjs here also?

Should this array be somewhere more global so it can be re-used, or does it belong here only?

Reply:
We previously used `require.extensions` which meant not needing to hard-code this. I'm not sure if anything replaced it.